platform_channels/lib/src/counter_method_channel.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+/// This class includes implementation of two platform methods [increment],
+/// and [decrement] which are used to increment and decrement value
+/// of count respectively.
+class Counter {
+  /// Creates a [MethodChannel] with the specified name to invoke platform method.
+  /// In order to communicate across platforms, the name of MethodChannel
+  /// should be same on native and dart side.
+  static MethodChannel methodChannel = const MethodChannel('methodChannelDemo');
+
+  /// This method is responsible to increment and return the value of count.
+  static Future<int> increment({int counterValue}) async {
+    try {
+      // Invokes a platform method with given name and argument. The first
+      // argument to MethodChannel.invokeMethod represents the name of the
+      // platform method. The second argument represents the arguments passed
+      // to platform method.
+      final result = await methodChannel
+          .invokeMethod<int>('increment', {'count': counterValue});
+      return result;
+    } on Exception catch (_) {
```

Comment:
nit: this is a style concern, and several of the other contributors to the repo disagree with me, but I generally avoid using underscores for unused parameter names. While they're perfectly fine in normal code, a reader here might like to know what data that parameter normally holds. Using a real name (even something like `ex` here) can give them a hint as they read.